components/frontend/src/App.test.js

```diff
@@ -69,32 +68,47 @@ it("resets the user when the user clicks logout", async () => {
     expect(screen.queryAllByText(/admin/).length).toBe(0)
 })
 
+async function selectDate() {
+    await act(async () => {
+        fireEvent.click(screen.getByLabelText("Report date"))
+    })
+    await act(async () => {
+        fireEvent.click(screen.getByRole("button", { name: "Previous month" }))
+    })
+    await act(async () => {
+        fireEvent.click(screen.getAllByRole("gridcell", { name: "15" })[0])
+    })
+}
+
 it("handles a date change", async () => {
     render(<App />)
-    await userEvent.type(screen.getByPlaceholderText("YYYY-MM-DD"), "2020-03-13")
-    expect(screen.getAllByDisplayValue("2020-03-13").length).toBe(1)
+    await selectDate()
+    expect(screen.getByLabelText("Report date").textContent).toMatch(/15/)
 })
 
 it("handles a date change between two dates in the past", async () => {
     history.push("/?report_date=2022-03-13")
     render(<App />)
-    await userEvent.type(screen.getByPlaceholderText("YYYY-MM-DD"), "{Backspace}4")
-    expect(screen.getAllByDisplayValue("2022-03-14").length).toBe(1)
+    await selectDate()
+    expect(screen.getByLabelText("Report date").textContent).toMatch(/15/)
```

Comment:
Does this test still cover its intended use? Perhaps we should check the month of the date?

Reply:
Yeah, since we now have dayjs as dependency I've added a calculation of the expected date to the test so we can match on the whole date instead of just the day of the month.